packages/qwik-city/runtime/src/image-component.tsx

```diff
@@ -0,0 +1,194 @@
+  style?: Record<string, string | number>;
+  aspectRatio?: number;
+  layout: 'fixed' | 'constrained' | 'fullWidth';
+  objectFit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down' | 'inherit' | 'initial';
+  resolutions?: number[];
+}
+
+export const ImageContext = createContextId<ImageState>('ImageContext');
+
+/**
+ * @alpha
+ */
+export const useImageProvider = (state: ImageState) => {
+  useContextProvider(ImageContext, state);
+};
+
+export const getStyles = ({
+  width,
+  height,
+  aspectRatio,
+  objectFit = 'cover',
+  layout,
+}: Pick<ImageProps, 'width' | 'height' | 'aspectRatio' | 'objectFit' | 'layout'>): Record<
+  string,
+  string | undefined
+> => {
+  const isValid = (value?: string | number) => value || value === 0;
+  const objectFitStyle = { 'object-fit': objectFit };
+  switch (layout) {
+    case 'fixed':
+      return {
+        ...objectFitStyle,
+        width: isValid(width) ? `${width}px` : undefined,
+        height: isValid(height) ? `${height}px` : undefined,
+      };
+    case 'constrained':
+      return {
+        ...objectFitStyle,
+        width: '100%',
+        'max-width': isValid(width) ? `${width}px` : undefined,
+        'max-height': isValid(height) ? `${height}px` : undefined,
+        'aspect-ratio': isValid(aspectRatio) ? `${aspectRatio}` : undefined,
+      };
+    case 'fullWidth':
+      return {
+        ...objectFitStyle,
+        width: '100%',
+        'aspect-ratio': isValid(aspectRatio) ? `${aspectRatio}` : undefined,
+        height: isValid(height) ? `${height}px` : undefined,
+      };
+  }
+};
+
+export const getSizes = ({ width, layout }: Pick<ImageProps, 'width' | 'layout'>) => {
+  if (!width || !layout) {
+    return undefined;
+  }
+  switch (layout) {
+    // If screen is wider than the max size, image width is the max size,
+    // otherwise it's the width of the screen
+    case `constrained`:
+      return `(min-width: ${width}px) ${width}px, 100vw`;
+
+    // Image is always the same width, whatever the size of the screen
+    case `fixed`:
+      return `${width}px`;
+
+    // Image is always the width of the screen
+    case `fullWidth`:
+      return `100vw`;
+
+    default:
+      return undefined;
+  }
+};
+
+export const getSrcSet = ({
+  src,
+  width,
+  height,
+  aspectRatio,
+  resolutions,
+  layout,
+}: Pick<
+  ImageProps,
+  'src' | 'width' | 'height' | 'aspectRatio' | 'resolutions' | 'layout'
+>): string => {
+  const breakpoints = getBreakpoints({
+    width: typeof width === 'string' ? parseInt(width, 10) : width,
+    layout,
+    resolutions: resolutions || DEFAULT_RESOLUTIONS,
+  });
+  return breakpoints
+    .sort()
+    .map((bp) => {
+      let transformedHeight;
+      if (height && aspectRatio) {
+        transformedHeight = Math.round(bp * aspectRatio);
+      }
+
+      const transformed = {
+        url: src,
+        width: bp,
+        height: transformedHeight,
+      };
+
+      return `${transformed.url}?w=${bp} ${bp}w`;
```

Comment:
The point of using a loader function here is so that we can plug in any image service. This becomes quite limiting.

Reply:
You're right, I will call a loader function here so everyone can create their own logic